tests/Backend/Mixed/SharingTest.php

```diff
@@ -1200,6 +1231,274 @@ function (array $column) {
+            $this->assertArrayHasKey('timestamp', $tableMetadata);
+
+            $this->assertEquals('test', $tableMetadata['provider']);
+            $this->assertEquals('test.metadata.key', $tableMetadata['key']);
+            $this->assertEquals('test.metadata.value', $tableMetadata['value']);
+
+            $columns = $table['columns'];
+
+            $this->assertSame(5, count($columns));
+            foreach ($columns as $column) {
+                $this->assertTrue(in_array($column, $tableColumns));
+            }
+
+            $columnsMetadata = reset($table['columnMetadata']['transid']);
+
+            $this->assertArrayHasKey('id', $columnsMetadata);
+            $this->assertArrayHasKey('key', $columnsMetadata);
+            $this->assertArrayHasKey('value', $columnsMetadata);
+            $this->assertArrayHasKey('provider', $columnsMetadata);
+            $this->assertArrayHasKey('timestamp', $columnsMetadata);
+
+            $this->assertEquals('test', $columnsMetadata['provider']);
+            $this->assertEquals('test.metadata.key', $columnsMetadata['key']);
+            $this->assertEquals('test.metadata.value', $columnsMetadata['value']);
+        }
+    }
+
+    private function sharedBucketDetailWithAliasTableMetadataTest($bucketId, $tableId, $project)
+    {
+        $columnId = $tableId . '.transid';
+        $mdList = $this->prepareTestMetadata($bucketId, $tableId, $columnId);
+
+        // create alias of alias
+        $this->_client->createAliasTable(
+            $bucketId,
+            $tableId,
+            'tableAlias'
+        );
+        $tableAliasAliasName = 'tableAliasAlias';
+        $this->_client->createAliasTable(
+            $bucketId,
+            $bucketId . '.tableAlias',
+            $tableAliasAliasName
+        );
+
+        $response = $this->_client2->getSharedBucketDetail(
+            $project['id'],
+            $bucketId,
+            [
+                'include' => 'columns,tableMetadata,columnMetadata'
+            ]
+        );
+
+        $aliasAliasTableId = $bucketId . '.' . $tableAliasAliasName;
+        $tables = array_values(array_filter($response['tables'], function ($table) use ($aliasAliasTableId) {
+            return $table['id'] === $aliasAliasTableId;
+        }));
+
+        // the metadata should be propagated from the source table
+        $this->assertNotEmpty($tables[0]['sourceTable']['columnMetadata']['transid']);
+        $this->assertEquals(
+            $mdList,
+            $tables[0]['sourceTable']['columnMetadata']['transid']
+        );
+
+        $this->assertArrayHasKey('metadata', $tables[0]);
+        $this->assertArrayHasKey('columns', $tables[0]);
+        $this->assertArrayHasKey('columnMetadata', $tables[0]);
+
+        $this->assertSame(0, count($tables[0]['metadata']));
+
+        $columns = $tables[0]['columns'];
+        $tableColumns = $this->_client->getTable($tableId)['columns'];
+        $this->assertSame(5, count($columns));
```

Comment:
tady jsi klidně mohl nechat přímo assert same na ty pole - měly by být ve stejném pořadí...

Reply:
Takhle to neručí, že sloupce jsou ve stejném pořadí...